diff --git a/fcos/core/csrc/cuda/ROIAlign_cuda.cu b/fcos/core/csrc/cuda/ROIAlign_cuda.cu
index 930e836..b8f8b21 100644
--- a/fcos/core/csrc/cuda/ROIAlign_cuda.cu
+++ b/fcos/core/csrc/cuda/ROIAlign_cuda.cu
@@ -2,7 +2,7 @@
 #include <ATen/ATen.h>
 #include <ATen/cuda/CUDAContext.h>
 
-#include <THC/THC.h>
+#include <c10/cuda/CUDAGuard.h>
 #include <THC/THCAtomics.cuh>
 #include <THC/THCDeviceUtils.cuh>
 
diff --git a/fcos/core/csrc/cuda/ROIPool_cuda.cu b/fcos/core/csrc/cuda/ROIPool_cuda.cu
index 85210fe..bb77a2f 100644
--- a/fcos/core/csrc/cuda/ROIPool_cuda.cu
+++ b/fcos/core/csrc/cuda/ROIPool_cuda.cu
@@ -2,7 +2,7 @@
 #include <ATen/ATen.h>
 #include <ATen/cuda/CUDAContext.h>
 
-#include <THC/THC.h>
+#include <c10/cuda/CUDAGuard.h>
 #include <THC/THCAtomics.cuh>
 #include <THC/THCDeviceUtils.cuh>
 
diff --git a/fcos/core/csrc/cuda/SigmoidFocalLoss_cuda.cu b/fcos/core/csrc/cuda/SigmoidFocalLoss_cuda.cu
index d8ca726..f409cb4 100644
--- a/fcos/core/csrc/cuda/SigmoidFocalLoss_cuda.cu
+++ b/fcos/core/csrc/cuda/SigmoidFocalLoss_cuda.cu
@@ -5,7 +5,7 @@
 #include <ATen/ATen.h>
 #include <ATen/cuda/CUDAContext.h>
 
-#include <THC/THC.h>
+#include <c10/cuda/CUDAGuard.h>
 #include <THC/THCAtomics.cuh>
 #include <THC/THCDeviceUtils.cuh>
 
diff --git a/fcos/core/csrc/cuda/deform_conv_cuda.cu b/fcos/core/csrc/cuda/deform_conv_cuda.cu
index 3ed7084..7115174 100644
--- a/fcos/core/csrc/cuda/deform_conv_cuda.cu
+++ b/fcos/core/csrc/cuda/deform_conv_cuda.cu
@@ -8,7 +8,7 @@
 #include <ATen/ATen.h>
 #include <ATen/cuda/CUDAContext.h>
 
-#include <THC/THC.h>
+#include <c10/cuda/CUDAGuard.h>
 #include <THC/THCDeviceUtils.cuh>
 
 #include <vector>
diff --git a/fcos/core/csrc/cuda/deform_pool_cuda.cu b/fcos/core/csrc/cuda/deform_pool_cuda.cu
index 554f9d4..ee43e62 100644
--- a/fcos/core/csrc/cuda/deform_pool_cuda.cu
+++ b/fcos/core/csrc/cuda/deform_pool_cuda.cu
@@ -12,7 +12,7 @@
 #include <ATen/ATen.h>
 #include <ATen/cuda/CUDAContext.h>
 
-#include <THC/THC.h>
+#include <c10/cuda/CUDAGuard.h>
 #include <THC/THCDeviceUtils.cuh>
 
 #include <vector>
diff --git a/fcos/core/csrc/cuda/ml_nms.cu b/fcos/core/csrc/cuda/ml_nms.cu
index d13d7df..b784464 100644
--- a/fcos/core/csrc/cuda/ml_nms.cu
+++ b/fcos/core/csrc/cuda/ml_nms.cu
@@ -2,7 +2,7 @@
 #include <ATen/ATen.h>
 #include <ATen/cuda/CUDAContext.h>
 
-#include <THC/THC.h>
+#include <c10/cuda/CUDAGuard.h>
 #include <THC/THCDeviceUtils.cuh>
 
 #include <vector>
diff --git a/fcos/core/csrc/cuda/nms.cu b/fcos/core/csrc/cuda/nms.cu
index 077faa8..34d0457 100644
--- a/fcos/core/csrc/cuda/nms.cu
+++ b/fcos/core/csrc/cuda/nms.cu
@@ -2,7 +2,7 @@
 #include <ATen/ATen.h>
 #include <ATen/cuda/CUDAContext.h>
 
-#include <THC/THC.h>
+#include <c10/cuda/CUDAGuard.h>
 #include <THC/THCDeviceUtils.cuh>
 
 #include <vector>
